### PR TITLE
Module-3: implement solution 

### PR DESCRIPTION
**Pull request by igorpolianskyi**

### CODE REVIEW

File: module_1/src/components/MovieTile/MovieTile.tsx [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+  const formatGenres = getFormatGenres(movie.genres);
+
+
+  const handleClick = () => {
+    onClick?.(movie.name);
+  };
+
+  const handleEdit = (e: React.MouseEvent) => {
+    e.stopPropagation();
+    console.log('handleEdit')
```
Comment (tishchenko-mykyta) [inline module_1/src/components/MovieTile/MovieTile.tsx:25]: remove logs

File: module_1/src/components/MovieDetails/MovieDetails.test.tsx [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+  it('renders movie year and duration', () => {
+    render(<MovieDetails movie={movie} />);
+    expect(screen.getByText(movie.year.toString())).toBeInTheDocument();
+    expect(screen.getByText(movie.duration)).toBeInTheDocument();
+  });
+
+  it('renders movie description', () => {
+    render(<MovieDetails movie={movie} />);
+    expect(screen.getByText(movie.description)).toBeInTheDocument();
+  });
```
Comment (tishchenko-mykyta) [inline module_1/src/components/MovieDetails/MovieDetails.test.tsx:45]: Minor
Edge case test with missing/empty description could be added

File: module_1/src/components/SortControl/SortControl.tsx [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+import styles from './SortControl.module.css'
+
+export type SortByOption = 'RELEASE DATE' | 'TITLE';
+
+interface SortControlProps {
+  value: SortByOption;
+  onSortChange: (newValue: SortByOption) => void;
+}
+
+const SortControl: React.FC<SortControlProps> = ({ value, onSortChange }) => {
```
Comment (tishchenko-mykyta) [inline module_1/src/components/SortControl/SortControl.tsx:11]: Looks like styles are a bit off 
Image

Design
Image

File: module_1/src/components/SearchForm/SearchForm.stories.tsx [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+
+export const Default: Story = {
+  args: {
+    onSearch: (query) => alert(`Searching for: ${query}`),
+  },
+};
+
+export const WithInitialQuery: Story = {
+  args: {
+    initialQuery: 'Inception',
```
Comment (tishchenko-mykyta) [inline module_1/src/components/SearchForm/SearchForm.stories.tsx:20]: Component is uncontrolled - manages its own state. 
The change of initialQuery in storybook does not change components state immediately, only after refresh
The same happens int the `Counter` compoent

Possible fixes:
Move query state to a parent:
```js
const [query, setQuery] = useState<string>(initialQuery);
```
Or sync when the prop changes:
```js
React.useEffect(() => {
    setQuery(initialQuery);
}, [initialQuery]);
```

**Review by tishchenko-mykyta — APPROVED** [review]
